app-frontend/src/app/core/services/imageOverlay.service.js

```diff
@@ -0,0 +1,107 @@
+/* eslint spaced-comment: 0 */
+
+export default (app) => {
+
+    class ImageOverlayService {
+
+        /** Returns a new Image overlay layer that clips to a datamask
+         *
+         * @param {string} url location to request tiles from
+         * @param {L.Bounds} bounds leaflet bounds for image
+         * @param {Object} options additional parameters to create image overlay
+         *
+         * @returns {L.Layer} image layer to put on map for thumbnails
+         */
+        createNewImageOverlay(url, bounds, options) {
+            let ImageOverlayLayer = L.ImageOverlay.extend({
+                options: {
+                    opacity: 1,
+                    alt: '',
+                    interactive: false,
+                    crossOrigin: false,
+                    dataMask: {},
+                    thumbnail: ''
+                },
+
+                _initImage: function () {
+                    this._image = L.DomUtil.create('div', 'leaflet-image-layer ' +
+                        (this._zoomAnimated ? 'leaflet-zoom-animated' : ''));
+                },
+
+                _reset: function () {
+                    let tile = this._image;
+                    let tileBounds = new L.Bounds(
+                        this._map.latLngToLayerPoint(this._bounds.getNorthWest()),
+                        this._map.latLngToLayerPoint(this._bounds.getSouthEast())
+                        );
+                    let size = this.size = tileBounds.getSize();
+
+                    L.DomUtil.setPosition(tile, tileBounds.min);
+
+                    tile.style.width = size.x + 'px';
+                    tile.style.height = size.y + 'px';
+
+                    let dataMask = this.options.dataMask;
+
+                    let result = dataMask.coordinates[0][0].map((lngLat) => {
+                        let point = this._map.latLngToLayerPoint([lngLat[1], lngLat[0]],
+                            this._map.getZoom()
+                        );
+                        return [point.x - tileBounds.min.x, point.y - tileBounds.min.y];
+                    });
+
+                    let points = _.flatten(result);
+
+
+                    let stage = new Konva.Stage({
+                        container: tile,
+                        width: size.x,
+                        height: size.y
+                    });
+
+                    let layer = this.layer = new Konva.Layer();
+
+                    let img = this.img = L.DomUtil.create('img');
+                    img.src = this.options.thumbnail.url;
+
+                    let lineConfig = {
+                        points: points,
+                        width: size.x,
+                        height: size.y,
+                        fillPriority: 'pattern',
+                        fillPatternImage: img,
+                        fillPatternRepeat: 'no-repeat',
+                        fillPatternX: 0,
+                        fillPatternY: 0,
+                        stroke: 'grey',
```

Comment:
Is there a reason for the stroke? I tested out setting stroke to `false` and that works nicely as well without any border on the image. It does seem like it might help hide any imperfections in the clipping.